Undefined name: Where is o_to_meta_xml() defined?

```
./acs4.py:180:31: F821 undefined name 'o_to_meta_xml'
                    meta_el = o_to_meta_xml(v)
                              ^
./acs4.py:582:16: F821 undefined name 'o_to_meta_xml'
        return o_to_meta_xml(o)
```
https://github.com/internetarchive/acs4_py/search?q=o_to_meta_xml&unscoped_q=o_to_meta_xml

--- a/acs4.py
+++ b/acs4.py
@@ -190,7 +190,7 @@ def request(server, api, action, request_args, password,
                     api_el.append(perms_el)
             elif key == 'metadata':
                 if isinstance(v, dict):
-                    meta_el = o_to_meta_xml(v)
+                    meta_el = o_to_meta_el(v)
                 else:
                     meta_el = read_xml(v, 'metadata')
                 api_el.append(meta_el)
@@ -592,7 +592,7 @@ def o_to_meta_el(o):
 
 def o_to_el(o, name):
     if name == 'metadata':
-        return o_to_meta_xml(o)
+        return o_to_meta_el(o)
     el = etree.Element(name)
     for k, v in o.iteritems():
         if isinstance(v, dict):

--- a/acs4.py
+++ b/acs4.py
@@ -7,21 +7,34 @@ queryresourceitems, upload, request and mint.
 """
 from __future__ import print_function
 
-import sys
-import re
-import os
-import math
-import httplib
-import urllib
-from lxml import etree
-from StringIO import StringIO
 import base64
-import hmac
-import hashlib
-import random
-import time
 import datetime
+import hashlib
+import hmac
+import math
+import os
+import random
+import re
+import sys
+import time
+import urllib
 import uuid
+
+from lxml import etree
+
+try:  # Python 3
+    import http.client as httplib
+    from io import StringIO
+except ImportError:  # Python 2
+    from StringIO import StringIO
+    import httplib
+
+try:
+    basestring
+    unicode
+except NameError:
+    basestring = str
+    unicode = str
 
 AdeptNS = 'http://ns.adobe.com/adept'
 AdeptNSBracketed = '{' + AdeptNS + '}'
@@ -98,7 +111,7 @@ def mint(server, secret, resource, action, ordersource, rights=None, orderid=Non
     urlargs = urllib.urlencode(argsobj)
     mac = hmac.new(base64.b64decode(secret), urlargs, hashlib.sha1)
     auth = mac.hexdigest()
-    portstr = (':' + str(port)) if port is not 80 else ''
+    portstr = '' if port == 80 else ':{}'.format(port)
 
     # replace with string format?
     # construct with urlparse.unsplit()

--- a/bss.py
+++ b/bss.py
@@ -1,20 +1,19 @@
 #!/usr/bin/env python
 
-#import cgitb; cgitb.enable()
-import sys
-
-import web
 import json
-
-import warnings
-warnings.filterwarnings("ignore", message="the sets module is deprecated")
-import MySQLdb
-
+import sys
 import uuid
+import warnings
+
+# import cgitb; cgitb.enable()
+import MySQLdb
+import web
+
+warnings.filterwarnings("ignore", message="the sets module is deprecated")
 
 
 #def cgidebugerror():
-#    """                                                                         
+#    """
 #    """
 #    _wrappedstdout = sys.stdout
 #    sys.stdout = web._oldstdout
@@ -22,7 +21,6 @@ import uuid
 #    sys.stdout = _wrappedstdout
 #
 #web.internalerror = cgidebugerror
-
 
 
 urls = (
@@ -36,7 +34,7 @@ urls = (
 
 app = web.application(urls, globals())
 
-dbschema = """ 
+dbschema = """
 
 mysql> show databases;
 +--------------------+
@@ -147,9 +145,9 @@ class acs4db():
 
     def get_fulfillment_info(self, resource=None):
         """ returns a list of resources in the fulfilment table , values set to dict of handy facts """
-        
-        resources = [] 
-        
+
+        resources = []
+
         if resource == '':
             resource = None
 
@@ -187,8 +185,8 @@ class acs4db():
 
     def get_loaned_out(self, resource=None):
         """ returns a list of unloanable books (someone else has 'em) according to acs"""
-        resources = [] 
-        
+        resources = []
+
         if resource == '':
             resource = None
 
@@ -201,9 +199,9 @@ class acs4db():
                             (
                                 (loanuntil IS NULL OR until IS NULL)
                                 OR
-                                (loanuntil > NOW()) 
+                                (loanuntil > NOW())
                             )
-                            AND 
+                            AND
                             ( returned IS NULL OR returned = 'F')
                         )
         """
@@ -272,7 +270,7 @@ class acs4db():
             #sys.stderr.write(r['resourceid'] + "\n")
             #json.dumps(r)
             r = self._fetchone_dict(c)
-            
+
 
         return resources
 
@@ -313,22 +311,22 @@ class acs4db():
             r['loanstatus'] = loanstatus
             resources.append(r)
             r = self._fetchone_dict(c)
-        
+
         return resources
 
     def get_transaction_info(self, transid):
-	sql = ("SELECT ri.identifier, fi.resourceid, f.transid, f.returned, f.transtime, f.loanuntil"  +
-	       " FROM fulfillmentitem fi, fulfillment f, resourceitem ri" + 
-	       " WHERE ri.resourceid=fi.resourceid and fi.fulfillmentid=f.fulfillmentid and f.transid=%s")
+        sql = ("SELECT ri.identifier, fi.resourceid, f.transid, f.returned, f.transtime, f.loanuntil"  +
+               " FROM fulfillmentitem fi, fulfillment f, resourceitem ri" +
+               " WHERE ri.resourceid=fi.resourceid and fi.fulfillmentid=f.fulfillmentid and f.transid=%s")
 
-	self.connect()
+        self.connect()
         c = self.conn.cursor()
         c.execute(sql, (transid, ))
-	row = self._fetchone_dict(c)
-	if row:
-	  row['resourceid'] = 'urn:uuid:' + str(uuid.UUID(bytes=row['resourceid']))
-	  row['loanuntil'] = row['loanuntil'].isoformat()
-	  row['transtime'] = row['transtime'].isoformat()
+        row = self._fetchone_dict(c)
+        if row:
+            row['resourceid'] = 'urn:uuid:' + str(uuid.UUID(bytes=row['resourceid']))
+            row['loanuntil'] = row['loanuntil'].isoformat()
+            row['transtime'] = row['transtime'].isoformat()
         return row
 
 class is_loaned_out:
@@ -364,20 +362,19 @@ class transaction_info:
 class item:
     def GET(self, identifier):
         web.header("Content-Type", 'text/plain')
-	db = acs4db()
-	d = {
-	  "identifier": identifier,
-	  "resources": [self.process_resource(db, x) for x in db.get_resource_info_by_id(identifier)]
-	}
-	return json.dumps(d, sort_keys=True, indent=4)
+        db = acs4db()
+        d = {
+          "identifier": identifier,
+          "resources": [self.process_resource(db, x) for x in db.get_resource_info_by_id(identifier)]
+        }
+        return json.dumps(d, sort_keys=True, indent=4)
 
     def process_resource(self, db, resource):
-	d = {}
-	for k in ['resourceid', 'src', 'format']:
-	  d[k] = resource[k]
-	d['loans'] = db.get_loaned_out(d['resourceid'])
-	return d
+        d = {}
+        for k in ['resourceid', 'src', 'format']:
+            d[k] = resource[k]
+        d['loans'] = db.get_loaned_out(d['resourceid'])
+        return d
 
 if __name__ == "__main__":
     app.run()
-

